inbm-v5.md

```diff
@@ -0,0 +1,320 @@
+        [--reboot, -rb; default=yes]
+        [--package-list, -p=PACKAGES]
+      ```
+
+    For detailed usage of `inbc` refer to ![inbc usage guide](inbc-readme-link)
+
+1. #### cloud-client
+
+   - **Function**: Cloud `device management service` (DMS) connecting thread
+   - **Main Tasks**:
+      - North-bound acts MQTT client connecting to DMS (e.g. Azure IOT Central or ThingsBoard)
+      - South-bound acts as `inbm-daemon` client translating over-the-air (ota) commands from DMS to `inbm-daemon` gRPC API's
+      - Checks on any `state` file to perform additional tasks on startup, e.g. post a OS update related bootup.
+
+1. #### dispatcher-queue
+
+   - **Function**: Management command queue
+   - **Main Tasks**:
+      - implements a simple queue of size `1` for device management commands
+      - invokes `updater` thread based on the type of update command e.g. firmware or os or application
+
+1. #### updater threads
+
+   - **Function**: A `transieant` thread performing update on underlying host
+   - **Main Tasks**:
+      - _Firmware updater_: Perfomrs firmware update related tasks like:
+         - check applicability, i.e. verdor, version and date checks
+         - download capsule file and perform signature checks if applicable
+         - invoke IBV's firmware update tool based on firmware-update config file look up.
+         - update logging and state files
+         - send intermediate results to `inbm-daemon` for reporting
+         - trigger reboot of platform if applicable
+      - _OS updater_: Perfomrs OS update related tasks like:
+         - check applicability, e.g. checks available disk space
+         - download OS image file and perform signature checks if applicable
+         - invoke OS update tool based on underlying OS type/distribution.
+         - update logging and state files
+         - send intermediate results to `inbm-daemon` for reporting
+         - trigger reboot of platform if applicable
+      - _Application updater_: Perfomrs application update related tasks like:
+         - check applicability, e.g. checks available disk space
+         - invoke underlying OS distributions `package manager` to perfomr the required installation tasks.
+         - update logging and state files
+         - send results to `inbm-daemon` for reporting
+
+1. #### telemerty-reporter
+
+   - **Function**: Thread performing basic platform telemetry collection and reporting
+   - **Main Tasks**: Basic plaform telemetry being collected by `telemetry-reporter` can be catogarized as `static` and `dynamic`
+      - _Static_: Information that remains same for the most part of the a devices life cycle (e.g. UUID, Serial number etc) or only changes on updates (e.g. Firmware version, OS version etc)
+      - _Dynamic_: Information which constantly changes and is ideal to be plotted on a `time-sereis` database (e.g. CPU usage, memory usage etc)
+
+## Data Flow
```

Comment:
Are we also keeping the scheduling aspect that we put into TC recently?

Reply:
Not planning to.